Enhancements_Fast_api/main.py

```diff
@@ -0,0 +1,248 @@
+        )
+
+    find_pokemon.name = pokemon.name
+    find_pokemon.type_1 = pokemon.type_1
+    find_pokemon.type_2 = pokemon.type_2
+    find_pokemon.total = pokemon.total
+    find_pokemon.hp = pokemon.hp
+    find_pokemon.attack = pokemon.attack
+    find_pokemon.defense = pokemon.defense
+    find_pokemon.sp_atk = pokemon.sp_atk
+    find_pokemon.sp_def = pokemon.sp_def
+    find_pokemon.speed = pokemon.speed
+    find_pokemon.generation = pokemon.generation
+    find_pokemon.legendary = pokemon.legendary
+
+    db.commit()
+    return find_pokemon
+
+
+@app.patch("/pokemon/{pokemon_id}", response_model=PokemonPostPatchPutOutputSchema)
+def update_Pokemon_Patch(
+    pokemon_id: int, pokemon: PokemonPatchInputSchema, db: Session = Depends(get_db)
+):
+    find_pokemon = (
+        db.query(models.PokemonData).filter(models.PokemonData.id == pokemon_id).first()
+    )
+    if not find_pokemon:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"Pokemon with id {pokemon_id} doesn't exist...",
+        )
+
+    for key, value in pokemon.dict(exclude_unset=True).items():
+        setattr(find_pokemon, key, value)
+
+    db.commit()
+    return find_pokemon
+
+
+@app.delete("/pokemon/{pokemon_id}", status_code=status.HTTP_204_NO_CONTENT)
+def delete_Pokemon(pokemon_id: int, db: Session = Depends(get_db)):
+    find_pokemon = (
+        db.query(models.PokemonData).filter(models.PokemonData.id == pokemon_id).first()
+    )
+
+    if not find_pokemon:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="Pokemon not found"
+        )
+
+    db.delete(find_pokemon)
+    db.commit()
+    return None
+
+
+@app.post("/pokemon/")
+def fetch_and_store(db: Session = Depends(get_db)):
+    response = requests.get("https://coralvanda.github.io/pokemon_data.json")
+    data = response.json()
+
+    print(f"Data fetched: {len(data)} entries")
+
+    # Fetch the current max ID in the database
+    max_id = db.query(func.max(models.PokemonData.id)).scalar() or 0
+    current_id = max_id + 1
+
+    # Prepare the bulk data mapping
+    pokemon_list = []
+    for pokemon in data:
+        # Map the API data fields to the database model fields
+        pokemon_dict = {
+            "id": current_id,
+            "name": pokemon["Name"],
+            "type_1": pokemon["Type 1"],
+            "type_2": pokemon.get("Type 2", None),  # Handle optional type_2 field
+            "total": pokemon["Total"],
+            "hp": pokemon["HP"],
+            "attack": pokemon["Attack"],
+            "defense": pokemon["Defense"],
+            "sp_atk": pokemon["Sp. Atk"],
+            "sp_def": pokemon["Sp. Def"],
+            "speed": pokemon["Speed"],
+            "generation": pokemon["Generation"],
+            "legendary": pokemon["Legendary"],
+        }
+        pokemon_list.append(pokemon_dict)
+        current_id += 1  # Increment the ID for the next Pokémon
```

Comment:
good idea!! This is an efficient way to increment

Reply:
Thank You Manish